Client/src/Pages/Monitors/CreateMonitor/index.jsx

```diff
@@ -40,6 +42,23 @@ const CreateMonitor = () => {
   const [https, setHttps] = useState(true);
   const [errors, setErrors] = useState({});
 
+  useEffect(() => {
+    if(monitorId){
+      const data = monitors.find((monitor) => monitor._id === monitorId);
+      if (!data) {
+        console.error("Error fetching monitor of id: " + monitorId);
+        navigate("/not-found", {replace: true});
+      }
+      const {name, ...rest} = data;   //data.name is read-only
+      if(rest.type == 'http')  rest.url = rest.url.substring(8, rest.url.length)
```

Comment:
I think a better approach here would be to create a [URL instance](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) and parse the host  from it. 

This current code will not parse the URL correctly if the monitor is http procotcol instead of https:

`https://www.google.com` -> `www.google.com`
`http://www.google.com` -> `ww.google.com`

Reply:
Thanks for identifying the issue! Creating a URL instance does seem like a better option, I'll make the required changes.